Makefile

```
@@ -0,0 +1,24 @@
+.SUFFIXES:
+
+CC = gcc
+CFLAGS = -Wall -Wextra -pedantic -fpic -std=gnu99 -D_GNU_SOURCE
+LDFLAGS = -lstatsdc
+
+all: build build/libstatsdc.so build/example
+
+build/statsdc.o: src/statsdc.c
```

I believe this will fail unless directory `build` exists beforehand. This should fix it:
```suggestion
build/statsdc.o: src/statsdc.c | build
```
(Same for the next two rules.)

You could also choose to ignore the risk.

---

Why the pipe?

---

Prerequisites listed at the right of the pipe are [order-only prerequisites](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html): this says, in this case, that we only care that the directory `build` exists; its timestamp does not matter.
The distinction is probably not crucially important here, but, without the pipe, this would mean that touching any file within `build` would make `build/statsdc.o` dirty again, even though only changes to `src/statsdc.c` should trigger a recompile.

---

Ah, it's a GNU Make only feature.

---

Correct.

---

That's not very portable. :p

---

It's also a simple way of correctly expressing the dependency.
Perhaps another neat way of making sure that `build` exists is to commit a file in there.

---

So I pushed a commit with the order-only prerequisite; my thinking is that since we use two GNU functions, asprintf and vasprintf, it's okay to assume GNU Make.  I'm not a big fan of order-only prerequisites, but since gcc does not have a `--output-dir` option, it seems like the least bad option.

---

That's a valid assumption, presuming it also implies we don't intend to ever implement non-GNU fallbacks.